### PR TITLE
WIP rough implementation of anchoring and inferred duration for parallel timelines

### PR DESCRIPTION
**Pull request by nickschot**
proof of concept/experimentation for #172 

TODO:
- [x] anchoring (center, end)
- [x] inferred duration from longest known duration in the parallel timeline
- [x] sort motion definitions in parallel timeline by presence of `duration` (be it implicit like a Spring, or explicit)
- [x] fill behavior (currently inferred durations are "fill")
- [x] sort by duration as well, so longest is 1st and can be used for `fill` reference
- [ ] decide about `fill` overriding defined durations on MotionDefinitions if they're not the longest animation.
- [x] tests
- [ ] do a 2nd (sort) pass after the springs are resolved (if there are multiple) so their durations are known and can be taken into account

## Example with the current API

The `border` animation will take the duration of the Spring movement animation, since that is the longest animation. The background animation keeps it duration as it has a more specific `anchor: center` set on it's MotionDefinition. It is centered to the middle of the duration of the entire parallel timeline.

```typescript
{
      timeline: {
        type: 'parallel',
        anchor: 'fill',
        animations: [
          {
            sprites: keptSprites,
            properties: {
              border: {},
            },
            timing: {
              behavior: new TweenBehavior(),
              duration: 100, // TODO: this is currently overwritten if "fill" is specified on the Timeline.
            },
          },
          {
            sprites: keptSprites,
            properties: {
              translateX: {},
              translateY: {},
              width: {},
              height: {},
            },
            timing: {
              behavior: new SpringBehavior(),
            },
          },
          {
            sprites: keptSprites,
            properties: {
              backgroundColor: {},
            },
            timing: {
              behavior: new TweenBehavior(),
              duration: 1000,
              anchor: 'center',
            },
          },
        ],
      },
    }
```

### CODE REVIEW

**Comment by nickschot** [comment]
Example with video:

https://user-images.githubusercontent.com/334789/215115944-0651f314-d84a-4899-bf5d-b7a7dabeafef.mov

```js
timeline: {
        type: 'parallel',
        animations: [
          {
            sprites: keptSprites,
            properties: {
              x: {},
              y: {},
              width: {},
              height: {},
            },
            timing: {
              behavior: new TweenBehavior(),
              duration: 2000,
            },
          },
          {
            sprites: keptSprites,
            properties: {
              backgroundColor: {},
            },
            timing: {
              behavior: new TweenBehavior(),
              duration: 300,
              anchor: 'center',
            },
          },
          {
            sprites: keptSprites,
            properties: {
              border: {},
            },
            timing: {
              behavior: new TweenBehavior(),
              anchor: 'fill',
            },
          },
        ],
      }
```

File: packages/boxel-motion-test-app/tests/unit/models/orchestration-test.ts [hunk 1]
```diff
@@ -322,4 +323,597 @@ module('Unit | Orchestration', function () {
...
+                width: '20px',
+                borderColor: 'rgba(255, 255, 255, 1)',
+                backgroundColor: 'rgba(255, 255, 255, 1)',
+              },
+            ],
+          ],
+        ])
+      );
+    });
+  });
```
Comment (backspace) [inline packages/boxel-motion-test-app/tests/unit/models/orchestration-test.ts:918]: I appreciate the clarity of these unit tests 🤩

**Review by backspace — APPROVED** [review]
🎉

**Comment by lukemelia** [comment]
Closing due to inactivity

**Pull request closed without merging**